src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/SignatureHelper.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+
+namespace System.Reflection.Emit
+{
+    // TODO: Only support simple signatures. More complex signatures will be added.
+    internal static class MetadataSignatureHelper
+    {
+        internal static BlobBuilder FieldSignatureEncoder(Type fieldType)
+        {
+            var fieldSignature = new BlobBuilder();
+
+            WriteSignatureTypeForReflectionType(new BlobEncoder(fieldSignature).FieldSignature(), fieldType);
+
+            return fieldSignature;
+        }
+
+        internal static BlobBuilder MethodSignatureEncoder(ParameterInfo[]? parameters, ParameterInfo? returnType, bool isInstance)
+        {
+            Type[]? typeParameters = null;
+            Type? typeReturn = null;
+
+            if (parameters != null)
+            {
+                typeParameters = Array.ConvertAll(parameters, parameter => parameter.ParameterType);
+            }
+
+            if (returnType != null)
+            {
+                typeReturn = returnType.ParameterType;
+            }
+
+            return MethodSignatureEncoder(typeParameters, typeReturn, isInstance);
+        }
+        internal static BlobBuilder MethodSignatureEncoder(Type[]? parameters, Type? returnType, bool isInstance)
+        {
+            // Encoding return type and parameters.
+            var methodSignature = new BlobBuilder();
+
+            ParametersEncoder _parEncoder;
+            ReturnTypeEncoder _retEncoder;
```

Comment:
```suggestion
            ParametersEncoder parEncoder;
            ReturnTypeEncoder retEncoder;
```
fields should not start with undescores